### PR TITLE
Add appointOracle RPC

### LINKED ISSUE #202: `DeFiCh/ain:1.7.x` RPCs implementation
Similar to https://github.com/DeFiCh/jellyfish/issues/48 This tracks all the RPC methods and categories being implemented in `jellyfish-api-core`. Please tag update this if you want to work on a feature.

~~As a final docker build is not produced please use the latest 1.7.x docker build for your `testcontainers`, https://github.com/DeFiCh/jellyfish/pull/184 for more information.~~

### `rpc_icxorderbook.cpp`
- [x] icx_createorder @surangap #398 
- [x] icx_makeoffer @surangap #399 
- [x] icx_closeoffer @surangap #400 
- [x] icx_submitdfchtlc @surangap #404 
- [x] icx_submitexthtlc @surangap #405 
- [x] icx_claimdfchtlc @surangap #406 
- [x] icx_closeorder @surangap #408 
- [x] icx_getorder @surangap #401 
- [x] icx_listorders @surangap #402 
- [x] icx_listhtlcs @surangap #407 

### `rpc_oracles.cpp`
- [x] appointoracle @jingyi2811 #336 
- [x] removeoracle @jingyi2811 #339 
- [x] updateoracle @jingyi2811  #341 
- [x] setoracledata @jingyi2811 #346
- [x] getoracledata @jingyi2811 #347
- [x] listoracles @jingyi2811 #345
- [x] listlatestrawprices @jingyi2811 #352 
- [x] getprice @jingyi2811 #355
- [x] listprices @jingyi2811 #356

### `spv_rpc.cpp`
- [x] spv_createhtlc @Jouzo #511
- [x] spv_claimhtlc @Jouzo #521
- [x] spv_gethtlcseed @Jouzo  #527
- [x] spv_refundhtlc @Jouzo #529
- [x] spv_listhtlcoutputs @Jouzo #581
- [x] spv_decodehtlcscript @Jouzo #513
- [x] spv_getnewaddress @Jouzo #495 
- [x] spv_getaddresspubkey @Jouzo #496
- [x] spv_listreceivedbyaddress @Jouzo #498
- [x] spv_sendtoaddress @Jouzo  #506
- [x] spv_refundhtlcall @cwkang1998 #1324

### PR DESCRIPTION
**Pull request by jingyi2811**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Add appointoracle rpc

#### Which issue(s) does this PR fixes?:

Fixes part of #202

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    ]
+
+    const { txid, vout } = await container.fundAddress(await container.getNewAddress(), 10)
+    const promise = client.oracle.appointOracle(await container.getNewAddress(), priceFeeds, {
+      weightage: 1,
+      utxos: [{ txid, vout }]
+    })
+
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'Test AppointOracleTx execution failed:\ntx not from foundation member\', code: -32600, method: appointoracle')
```
Comment (monstrobishi) [inline packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts:66]: You can try checking the data created using `getoracledata` e.g.:

```
// Ensure oracle is updated and has correct values
    const getOracleDataResult = await container.call('getoracledata', [oracleId])
    expect(getOracleDataResult.priceFeeds.length).toStrictEqual(2)
    expect(getOracleDataResult.priceFeeds[0].token).toStrictEqual('APPLE')
    expect(getOracleDataResult.priceFeeds[0].currency).toStrictEqual('EUR')
```

File: packages/jellyfish-api-core/src/category/oracle.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ */
+export class Oracle {
+  private readonly client: ApiClient
+
+  constructor (client: ApiClient) {
+    this.client = client
+  }
+
+  /**
+   * Creates an oracle appointment transaction and saves it to the database.
```
Comment (monstrobishi) [inline packages/jellyfish-api-core/src/category/oracle.ts:14]: I know we are copying AIN documentation here, but I think we can do better, we can write something more descriptive like "Creates a price oracle for rely of real time price data".

**Review by monstrobishi — CHANGES_REQUESTED** [review]

File: packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts [hunk 3]
```diff
@@ -23,10 +23,26 @@ describe('Oracle', () => {
...
+    expect(txid.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const result = await container.call('getoracledata', [txid])
+    expect(result).toStrictEqual(
+      {
+        weightage: 1,
+        oracleid: txid,
+        address: expect.any(String), // This is an oracle address, not the address in line 26
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts:38]: I assume this is an oracle address. Kindly tell me if my assumption is wrong.

File: packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      { currency: 'USD', token: 'TESLA' },
+      { currency: 'EUR', token: 'APPLE' }
+    ]
+
+    const txid = await client.oracle.appointOracle(await container.getNewAddress(), priceFeeds, { weightage: 1 })
+
+    expect(typeof txid).toStrictEqual('string')
+    expect(txid.length).toStrictEqual(64)
+
+    await container.generate(1)
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts:31]: I need to add this line. If not,` Error: DeFiDRpcError: 'oracle <dc917a34a55be60761e74637d0d99005b7b2a6bc6a4a8acba6a2e6a3169b67d2> not found', code: -20
` is thrown.

File: packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts (hunk 3 above)
Comment (monstrobishi) [reply packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts:38]: This check is sufficient for now, as the getoracledata rpc call is returning the script hex instead of the address

**Review by monstrobishi — APPROVED** [review]

File: packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts (hunk 3 above)
Comment (jingyi2811) [reply packages/jellyfish-api-core/__tests__/category/oracle/appointOracle.test.ts]: Ok, then I remove the wrong comment

**Comment by jingyi2811** [comment]
Ahmed has approved this PR. But I remove the wrong comments after that.

**Review by fuxingloh — APPROVED** [review]
Should test a few more edge conditions, I will merge first since I can see you wan to avoid conflicts as you have many staged commits.
